packages/types/src/index.ts

```diff
@@ -619,3 +619,8 @@ export type TimingInterface = {
   markTime: (category: TimingCategory, label: string) => void;
   getTimingData: () => [string, string, number][];
 };
+
+export interface CodedError extends Error {
+  code?: number;
+  codeName?: string;
+}
```

Comment:
Just something to consider, the `MongoServerError` class from the driver package also describes an `Error` subclass with these optional properties (and more), and most of the time that's the kind of error for which we'd access the `.codeName` property, so it might make sense to use that type instead

Reply:
looks like the most generic `MongoError` only has a code, not a codeName https://github.com/mongodb/node-mongodb-native/blob/9401d09af4def8bfbeff65e70863be5d13b3dd61/src/error.ts#L123
But I could use the more specific `MongoServerError` type.